pycbc/detector/space.py

```diff
@@ -0,0 +1,899 @@
+# -*- coding: UTF-8 -*-
+#
+# =============================================================================
+#
+#                                   Preamble
+#
+# =============================================================================
+#
+"""
+This module provides utilities for simulating the GW response of space-based
+observatories.
+"""
+from abc import ABC, abstractmethod
+from pycbc.coordinates.space import TIME_OFFSET_20_DEGREES
+from pycbc.types import TimeSeries
+import numpy
+from numpy import cos, sin
+from astropy import constants
+import logging
+
+def get_available_space_detectors():
+    """List the available space detectors"""
+    dets = list(_space_detectors.keys())
+    aliases = []
+    for i in dets:
+        aliases.extend(_space_detectors[i]['aliases'])
+    return dets + aliases
+
+def parse_det_name(detector_name):
+    """Parse a string into a detector name and TDI channel.
+       The input is assumed to look like '{detector name}_{channel name}.'"""
+    out = detector_name.split('_', 1)
+    det = out[0]
+    try:
+        chan = out[1]
+    except IndexError:
+        # detector_name is just the detector, so save channel name as None
+        chan = None
+    return det, chan
+
+def apply_polarization(hp, hc, polarization):
+    """
+    Apply polarization rotation matrix.
+
+    Parameters
+    ----------
+    hp : array
+        The plus polarization of the GW.
+
+    hc : array
+        The cross polarization of the GW.
+
+    polarization : float
+        The SSB polarization angle of the GW in radians.
+
+    Returns
+    -------
+    (array, array)
+        The plus and cross polarizations of the GW rotated by the
+        polarization angle.
+    """
+    cphi = cos(2*polarization)
+    sphi = sin(2*polarization)
+
+    hp_ssb = hp*cphi - hc*sphi
+    hc_ssb = hp*sphi + hc*cphi
+
+    return hp_ssb, hc_ssb
+
+def preprocess(hp, hc, orbit_start_time, orbit_end_time,
```

Comment:
This might need a better name. Maybe it would be clearly by not mixing up applying polarization and checking the time span ?